scripts/publish-gh-pages.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable require-jsdoc */
+const { execSync } = require('child_process');
+const path = require('path');
+const github = require('@actions/github');
+const core = require('@actions/core');
+const githubToken = process.env.TOKEN;
+const userName = core.getInput('user_name') || process.env.GITHUB_ACTOR;
+const email = core.getInput('user_email') || `${process.env.GITHUB_ACTOR}@users.noreply.github.com`;
+const deployCommitMessage = `deploy: ${process.env.GITHUB_SHA}`;
+const worspaceDir = process.env.WORKSPACE_DIR;
+const docsDir = path.join(worspaceDir, 'docs');
+const docsBuildDir = path.join(worspaceDir, '../public');
+// TODO: change to `gh-pages` before merge
+const GH_PAGES_BRANCH = 'temp-gh-pages';

```

Comment:
I need to change this before mergin the PR with `gh-pages`.
For the test it will commit into the `temp-gh-pages` branch